fdbserver/masterserver.actor.cpp

```diff
@@ -1221,6 +1223,49 @@ ACTOR Future<Void> provideVersions(Reference<MasterData> self) {
 	}
 }
 
+void updateLiveCommittedVersion(Reference<MasterData> self, ReportRawCommittedVersionRequest req) {
+	self->minKnownCommittedVersion = std::max(self->minKnownCommittedVersion, req.minKnownCommittedVersion);
+	if (req.version > self->liveCommittedVersion.get()) {
+		self->databaseLocked = req.locked;
+		self->proxyMetadataVersion = req.metadataVersion;
+		// Note the set call switches context to any waiters on liveCommittedVersion before continuing.
+		self->liveCommittedVersion.set(req.version);
+	}
+	++self->reportLiveCommittedVersionRequests;
+	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR && req.writtenTags.present()) {
+		// NB: this if-condition is not needed after wait-for-prev is ported to this branch
+		if (req.version > self->ssVersionVector.maxVersion) {
```

Comment:
The if-condition is not needed. However, I think `self->ssVersionVector.setVersions(...` should be done before `self->liveCommittedVersion.set(req.version);`.

Reply:
(missed removing that condition, thanks!)
My intuition is also that `setVersions` should be before `liveCommittedVersion.set` as we don't want a context switch between the two operations and I will move it (I've not worked out how that could break things though).

Reply:
There is not wait() statement between these operations so how can context switch happen here?

Reply:
Oh, I see the comment above "self->liveCommittedVersion.set()".